airbyte-workers/src/main/java/io/airbyte/workers/storage/DockerComposeDocumentStoreClient.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.storage;
+
+import io.airbyte.commons.io.IOs;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Document store for when all we have is an FS. This should only be used in the docker-compose
+ * case. Leverages the workspace mount as a storage area.
+ */
+public class DockerComposeDocumentStoreClient implements CloudDocumentStoreClient {
+
+  private static final Path STATE_PATH = Path.of("state");
+  private final Path workspaceMount;
+
+  public DockerComposeDocumentStoreClient(final Path workspaceMount) {
+    this.workspaceMount = workspaceMount;
+  }
+
+  private Path getRoot() {
+    return workspaceMount.resolve(STATE_PATH);
+  }
+
+  private Path getPath(final String id) {
+    return getRoot().resolve(String.format("%s.yaml", id));
+  }
+
+  @Override
+  public void write(final String id, final String document) {
+    final Path path = getPath(id);
+    createDirectoriesSafe(path.getParent());
+    IOs.writeFile(path, document);
+  }
+
+  private void createDirectoriesSafe(final Path path) {
+    try {
+      Files.createDirectories(path);
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Optional<String> read(final String id) {
```

Comment:
About this prototype, why is it returning a String? I feel that we should return an `Optional<T>` and enforce the object is serialized with jackson (for example). I got a similar comments for the write. Is there any formart we want to use for the state than json. I mentioned Jackson because temporal is using it and the new state object are compatible with it.

Reply:
I'm pretty ambivalent. Pretty much if we ever want to store anything that's not json this might be a headache.

Reply:
I feel that it will be the opposite. The serialization will depend on this wrapper only and be shared between all the application that uses it. Could it be solve by using an interface that will specify a serialization type and use the serialization type to serialize an object. I feel that if we switch to a binary serialization (like protobuff) the current interface won't feet our needs.

Reply:
How soon do you think we'll need more specific type casting? 

If it's more than 6 months away, I favor leaving this as is now and adding this later. I'm not against it, but I value the simplicity/flexibility in the mean time to give us the most space to change things.

Reply:
I have an open (draft) PR that will need it. It is passing a `JobRunConfig` as an input to an activity. I was thinking about this usecase for this comment.

Reply:
It seems like it makes sense that we would use String and then have a wrapped version that lets us do Json storage with Jackson serialization.